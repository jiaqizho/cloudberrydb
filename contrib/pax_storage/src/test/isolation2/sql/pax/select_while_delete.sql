-- @Description Ensures that a select during a delete operation is ok
-- 
DROP TABLE IF EXISTS pax_tbl;
CREATE TABLE pax_tbl (a INT);
insert into pax_tbl select generate_series(1,100);

2: BEGIN;
2: SELECT * FROM pax_tbl WHERE a < 5 ORDER BY a;
2: DELETE FROM pax_tbl WHERE a < 5;
1: SELECT * FROM pax_tbl WHERE a >= 5 AND a < 10 ORDER BY a;
3: SELECT * FROM pax_tbl WHERE a < 5 ORDER BY a;
2: COMMIT;
2: SELECT * FROM pax_tbl WHERE a < 5 ORDER BY a;
4: SELECT * FROM pax_tbl WHERE a < 10 ORDER BY a;
4: INSERT INTO pax_tbl VALUES (0);
