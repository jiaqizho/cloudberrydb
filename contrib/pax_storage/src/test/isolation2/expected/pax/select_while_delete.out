-- @Description Ensures that a select during a delete operation is ok
--
DROP TABLE IF EXISTS pax_tbl;
DROP
CREATE TABLE pax_tbl (a INT);
CREATE
insert into pax_tbl select generate_series(1,100);
INSERT 100

2: BEGIN;
BEGIN
2: SELECT * FROM pax_tbl WHERE a < 5 ORDER BY a;
 a 
---
 1 
 2 
 3 
 4 
(4 rows)
2: DELETE FROM pax_tbl WHERE a < 5;
DELETE 4
1: SELECT * FROM pax_tbl WHERE a >= 5 AND a < 10 ORDER BY a;
 a 
---
 5 
 6 
 7 
 8 
 9 
(5 rows)
3: SELECT * FROM pax_tbl WHERE a < 5 ORDER BY a;
 a 
---
 1 
 2 
 3 
 4 
(4 rows)
2: COMMIT;
COMMIT
2: SELECT * FROM pax_tbl WHERE a < 5 ORDER BY a;
 a 
---
(0 rows)
4: SELECT * FROM pax_tbl WHERE a < 10 ORDER BY a;
 a 
---
 5 
 6 
 7 
 8 
 9 
(5 rows)
4: INSERT INTO pax_tbl VALUES (0);
INSERT 1
